In [22]:
import pandas as pd
import numpy as np
from scipy.stats import t 
import math


Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [23]:
s = 16
m = 80
n = 256
A = 0.05

In [24]:
st = s/(n ** 0.5)
st


1.0

In [25]:
a = norm.interval(1 - A, loc = m, scale = st)
a

(78.04003601545995, 81.95996398454005)

(78.04003601545995, 81.95996398454005)

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [26]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = len(x)
mean_ar = x.mean()
std_ar = np.std(x, ddof=1)


In [27]:
crit = abs(t.pdf(0.05/2 , (n-1)))


In [28]:
gr = crit * std_ar/np.sqrt(n)
a1 = mean_ar + gr
a1


6.645297513642881

In [29]:
b1 = mean_ar - gr
b1

6.53470248635712

6.645297513642881, 6.53470248635712


Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170
Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175

Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [30]:
daughters = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
mothers = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
n1 = len(daughters)
n2 = len(mothers)
mean1 = np.mean(daughters)
mean2 = np.mean(mothers)
std1 = np.std(daughters, ddof=1)
std2 = np.std(mothers, ddof=1)
alpha = 0.05
df = n1 + n2 - 2
t_critical = abs(t.ppf(alpha/2, df))

In [31]:
std_res = np.sqrt(std1**2/n1 + std2**2/n2)
margin_res = t_critical * std_res
a2 = mean1 - mean2 - margin_res
a2


-10.068418034506857

In [32]:
b2 = mean1 - mean2 + margin_res
b2

6.268418034506846

-10.068418034506857, 6.268418034506846